pyomo/contrib/mindtpy/iterate.py

```diff
@@ -22,15 +25,20 @@ def MindtPy_iteration_loop(solve_data, config):
             % solve_data.mip_iter)
 
         if algorithm_should_terminate(solve_data, config, check_cycling=False):
+            cuts_add_last_iteration = True
             break
 
         solve_data.mip_subiter = 0
         # solve MILP master problem
-        if config.strategy == 'OA':
+        if config.strategy in {'OA', 'GOA'}:
             master_mip, master_mip_results = solve_OA_master(
                 solve_data, config)
             if master_mip_results.solver.termination_condition is tc.optimal:
                 handle_master_mip_optimal(master_mip, solve_data, config)
+            elif master_mip_results.solver.termination_condition is tc.infeasible:
+                handle_master_mip_infeasible(master_mip, solve_data, config)
+                cuts_add_last_iteration = True
```

Comment:
The name of this option sounds weird. Maybe modify it to `last_iter_cuts`, since we assert that if this is false, we need to add them. Now it is more like a question (add cuts in the last iteration?) which does not correspond to the logic below.